trackpoint/src/module.c

```diff
@@ -29,6 +29,26 @@ void Module_Init(void)
     PORT_SetPinConfig(PS2_DATA_PORT, PS2_DATA_PIN,
                       &(port_pin_config_t){/*.pullSelect=kPORT_PullDown,*/ .mux=kPORT_MuxAsGpio});
     GPIO_PinInit(PS2_DATA_GPIO, PS2_DATA_PIN, &(gpio_pin_config_t){.pinDirection=kGPIO_DigitalInput, .outputLogic=0});
+
+    CLOCK_EnableClock(TP_RST_CLOCK);
+    PORT_SetPinConfig(TP_RST_PORT, TP_RST_PIN, &(port_pin_config_t){.pullSelect=kPORT_PullDown, .mux=kPORT_MuxAsGpio});
+    GPIO_PinInit(TP_RST_GPIO, TP_RST_PIN, &(gpio_pin_config_t){.pinDirection=kGPIO_DigitalOutput, .outputLogic=1});
+}
+
+bool shouldReset = false;
+
+void resetBoard()
+{
+    GPIO_WritePinOutput(TP_RST_GPIO, TP_RST_PIN, 0);
+
+    for (volatile uint32_t i=0; i<10000; i++);
```

Comment:
We'll optimize the firmware for low power consumption, especially for wireless, and busy loops hurt battery life, so please let's not use them, even if they're rarely invoked. Let's rather use the state machine to pull TP_RST low and then high.